### PR TITLE
feat: add `server.address` to db clients

### PR DESCRIPTION
**Pull request by quinna-h**
This is the first part of the effort to add the `server.address` tag to database clients. Every db span _must_ have a `server.address` tag based on the [OpenTelemetry definition](https://opentelemetry.io/docs/specs/semconv/general/attributes/#server-attributes), as per convention [for database spans](https://opentelemetry.io/docs/specs/semconv/database/database-spans/).

On the client side, it should match the remote service name; on the server side it should represent the local service name as seen externally on clients.

This modifies the following:

- aiomysql
- asyncpg
- cassandra
- django
- elasticsearch
- flask (cache)
- mariadb
- mysql
- mysqldb
- pymongo
- pymemcache
- pymysql
- snowflake
- sqlalchemy
- vertica

The new spans:
Screenshot 2024-08-07 at 12 43 38 PM

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: releasenotes/notes/feat-server-address-db-client-4e6af4ec4c4ce8ed.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Adds `server.address` tag to database client integrations to comply with OpenTelemetry standards.
```
Comment (emmettbutler) [inline releasenotes/notes/feat-server-address-db-client-4e6af4ec4c4ce8ed.yaml:4]:
```suggestion
    DBM: Adds `server.address` tag to database client integrations to comply with OpenTelemetry standards.
```

**Review by emmettbutler — APPROVED** [review]

**Review by mabdinur — COMMENTED** [review]
Overall looks good.

Is this a change we plan to make to all client libraries or is this a one off for dd-trace-py? If we plan to change this tag in all libraries it may be easier to do this remapping in trace-intake.

Also with this change `net.TARGET_HOST` and `net.SERVER_ADDRESS` will send duplicate data to the backend. If the goal is to standardize on `net.SERVER_ADDRESS` we should create a plan to deprecate `net.TARGET_HOST`.

File: ddtrace/contrib/cassandra/session.py [hunk 2]
```diff
@@ -258,12 +258,14 @@ def _extract_result_metas(result):
...
             host, _, port = host.partition(":")
             metas[net.TARGET_HOST] = host
+            metas[net.SERVER_ADDRESS] = host
             if port:
                 metas[net.TARGET_PORT] = int(port)
         elif hasattr(future, "_current_host"):
             address = deep_getattr(future, "_current_host.address")
             if address:
                 metas[net.TARGET_HOST] = address
+                metas[net.SERVER_ADDRESS] = host
```
[the remark this answers is not in the document]
Comment (quinna-h) [reply ddtrace/contrib/cassandra/session.py:268]: Good catch, thank you

File: ddtrace/contrib/internal/aiomysql/patch.py [hunk 3]
```diff
@@ -52,7 +53,6 @@ async def patched_connect(connect_func, _, args, kwargs):
 
     c = AIOTracedConnection(conn)
     Pin(tags=tags).onto(c)
```
Comment (mabdinur) [inline ddtrace/contrib/internal/aiomysql/patch.py:55]: do we need this?
```suggestion
    Pin(tags=tags).onto(c)
    return c
```

**Review by mabdinur — APPROVED** [review]

**Pull request merged**